### PR TITLE
Improve Dialog API

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
cc @sdroege @EPashkin @philn

### CODE REVIEW

File: Gir.toml [hunk 1]
```diff
@@ -758,6 +758,12 @@ manual_traits = ["DialogExtManual"]
     name = "get_content_area"
         [object.function.return]
         nullable = false
+    [[object.function]]
+    name = "run"
+    ignore = true
+    [[object.function]]
+    name = "get_response_for_widget"
+    ignore = true
```
Comment (sdroege) [inline Gir.toml]: Don't we have a way to override the return type?

**Comment by sdroege** [comment]
Good otherwise

File: Gir.toml (hunk 1 above)
Comment (GuillaumeGomez) [reply Gir.toml]: Doesn't seem like we do. :-/

**Comment by EPashkin** [comment]
👍

**Comment by GuillaumeGomez** [comment]
I need to update gtk examples as well.

File: src/tree_sortable.rs [hunk 2]
```diff
@@ -49,6 +51,15 @@ impl FromGlib<i32> for SortColumn {
     }
 }
 
+impl fmt::Display for SortColumn {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "SortColumn::{}", match *self {
+            SortColumn::Default => "Default",
+            SortColumn::Index(_) => "Index",
```
Comment (EPashkin) [inline src/tree_sortable.rs]: Don't print column number?

Comment (GuillaumeGomez) [reply src/tree_sortable.rs]: `Debug` does it. :)

Comment (sdroege) [reply src/tree_sortable.rs]: This is not a `Display` impl anyway but you know my opinion about that ;)

In this case it should print things like "Default column", "Column at index {}", etc. and not basically the same as the `Debug` impl.

Comment (GuillaumeGomez) [reply src/tree_sortable.rs]: Yes I know. But the current way of implementing `Display` is this one. However I agree this is not very nice... I'm open to suggestions for next release!

Comment (sdroege) [reply src/tree_sortable.rs]: My suggestion is that you can't automatically implement `Display` as it requires context specific knowledge about the type that we don't have.

For this specific type a correct `Display` implementation would be the one I mentioned.

**Comment by GuillaumeGomez** [comment]
CI passed so I merge.

About the `Display` implementations issue: we need to make something official. I don't fully agree with @sdroege's views on it and I think that @EPashkin might have some ideas about it. I'll open the issue on gir directly.

**Pull request merged**